### ISSUE TITLE
For the losses in your code 

### ISSUE THREAD

**Issue by themis0888** [bnulihaixia/Deep_metric#23]
Thanks for sharing your great work! 
I read some questions and your answers about the losses, but I ask again to make it sure.  
For the losses in your code, I see a similar type of losses such as 
neg_loss = 2.0/self.alpha * torch.log(1 + torch.sum(torch.exp(self.alpha * (neg_pair - base))))
from the SemiHard loss, DistWeighted loss. As far as I know, those losses were a linear loss of the distance(in your code, similarity) according to the original papers. (If those losses are from the FaceNet and Sampling matteres...)
Are these losses your own 'Weight' loss which has an assumption that the data follows the distribution of the mixture of Gaussian? 
Thanks :D

**Comment by bnulihaixia** [comment]
No such  assumption is needed.  Weight loss  is based on th idea of hard mining : to focus on harder samples, no matter positive or negative pairs.

**Comment by themis0888** [comment]
@bnulihaixia Thanks for your response! 
So, the loss in the DistWeight.py
neg_loss = 2.0/self.alpha * torch.log(1 + torch.sum(torch.exp(self.alpha * (neg_pair - base))))
this loss is a different one with the model in 'Sampling matters in deep embedding learning', right? 
In addition, according to your code, the default 'base' value in the loss is 0.5 now, is this a hyperparameter?  
Thanks.

**Comment by kalilili** [comment]
Sorry, in my opinion, this great performance has been gotten from L2 norm and parameter in your loss. The weighted way like your code has been explored by many papers, such as i.e. log(sum( exp(xxx)) or log(1 + sum(exp(xxx)) . This kind of weighted measure provide a grad proportion pattern of exp(dis1) : exp(dis2) : exp(dis3).
